controllers/application_controller.go

```diff
@@ -87,22 +87,31 @@ func (r *ApplicationReconciler) updateComponents(ctx context.Context, app *appv1
 }
 
 func (r *ApplicationReconciler) getNewApplicationStatus(ctx context.Context, app *appv1beta1.Application, resources []*unstructured.Unstructured, err error) *appv1beta1.ApplicationStatus {
-	objectStatuses := r.objectStatuses(ctx, resources)
+	var errs []error
+	if err != nil {
+		errs = append(errs, err)
+	}
+	objectStatuses, err := r.objectStatuses(ctx, resources)
+	if err != nil {
+		errs = append(errs, err)
+	}
 	aggReady, countReady := aggregateReady(objectStatuses)
 
 	newApplicationStatus := app.Status.DeepCopy()
 	newApplicationStatus.ComponentList = appv1beta1.ComponentList{
 		Objects: objectStatuses,
 	}
 	newApplicationStatus.ComponentsReady = fmt.Sprintf("%d/%d", countReady, len(objectStatuses))
-	if aggReady {
+	if aggReady && len(errs) == 0 {
 		setReadyCondition(newApplicationStatus, "ComponentsReady", "all components ready")
 	} else {
 		setNotReadyCondition(newApplicationStatus, "ComponentsNotReady", "some components not ready")
 	}
 
-	if err != nil {
-		setErrorCondition(newApplicationStatus, "ErrorSeen", err.Error())
+	if len(errs) != 0 {
+		for _, err := range errs {
+			setErrorCondition(newApplicationStatus, "ErrorSeen", err.Error())
```

Comment:
This will keep overwriting the error condition.
Maybe consider aggregating the errors.

Reply:
Will aggregate the errors in a separate PR. This PR is aiming to only change the way how do we compute status.